optuna/_transform.py

```diff
@@ -114,6 +121,13 @@ def transform(self, params: Dict[str, Any]) -> numpy.ndarray:
                 )
                 bound_idx += 1
 
+        if self._transform_0_1:
+            single_mask = self._raw_bounds[:, 0] == self._raw_bounds[:, 1]
+            trans_params[single_mask] = 0.5
+            trans_params[~single_mask] = (
+                trans_params[~single_mask] - self._raw_bounds[~single_mask, 0]
+            ) / (self._raw_bounds[~single_mask, 1] - self._raw_bounds[~single_mask, 0])
+
```

Comment:
I may have noticed another problem. Isn't this logic broken when `CategoricalDistribution` is contain?

How about moving this logic to the `_transform_numerical_param` and `_untransform_numerical_param` functions, like the `transform_log` argument?

Reply:
No, the `_raw_bounds` is the bounds of one-hot encoded values.

Reply:
I see. Does it sound reasonable to move this logic into `_transform_numerical_param` and `_untransform_numerical_param` like the `transform_log` argument?

Reply:
If you have categorical distribution with only one choice, I think it's more natural for that dimension to be 0.5, rather than 1. So I think it's more natural to add it as a "postprocess", not in `_transform_numerical_param`.

Reply:
For transformation, we not only need the value of the parameter but also the (transformed) bounds of that parameter. Currently, `_transform_numerical_param` and `_untransform_numerical_param` do not have the bounds as the argument, so it was a little inconvenient to implement it inside those functions.

Reply:
I see. Can we write it like below?

```python
    if transform_0_1 and d.single():
        trans_param = 0.5
    elif transform_0_1:
        low = d.low if transform_log and d.log else math.log(d.low)
        high = d.high if transform_log and d.log else math.log(d.high)
        trans_param = (trans_param - low) / (high - low)
```